src/ui/public/vislib/visualizations/_point_series_chart.js

```diff
@@ -1,91 +1,66 @@
 import d3 from 'd3';
 import _ from 'lodash';
-import VislibVisualizationsChartProvider from 'ui/vislib/visualizations/_chart';
-import VislibComponentsTooltipProvider from 'ui/vislib/components/tooltip';
+import ChartProvider from 'ui/vislib/visualizations/_chart';
+import TooltipProvider from 'ui/vislib/components/tooltip';
 import errors from 'ui/errors';
 
 export default function PointSeriesChartProvider(Private) {
 
-  const Chart = Private(VislibVisualizationsChartProvider);
-  const Tooltip = Private(VislibComponentsTooltipProvider);
+  const Chart = Private(ChartProvider);
+  const Tooltip = Private(TooltipProvider);
   const touchdownTmpl = _.template(require('ui/vislib/partials/touchdown.tmpl.html'));
 
   class PointSeriesChart extends Chart {
     constructor(handler, chartEl, chartData) {
       super(handler, chartEl, chartData);
     }
 
-    _stackMixedValues(stackCount) {
-      let currentStackOffsets = [0, 0];
-      let currentStackIndex = 0;
-
-      return function (d, y0, y) {
-        const firstStack = currentStackIndex % stackCount === 0;
-        const lastStack = ++currentStackIndex === stackCount;
+    getStackedCount() {
+      return _.reduce(this.handler.data.get('series'), function (sum, val) {
+        if (val.stacked || 1) return sum + 1;
+      }, 0);
+    };
 
-        if (firstStack) {
-          currentStackOffsets = [0, 0];
-        }
+    getStackedNum(data) {
+      let i = 0;
+      for (const seri of this.handler.data.get('series')) {
+        if (seri === data) return i;
+        if (seri.stacked || 1) i++;
```

Comment:
omg im so ashamed :) i hope i was at least super dr*nk when i was writing this ...